internal/http/services/owncloud/ocdav/spacelookup/spacelookup.go

```diff
@@ -105,7 +105,13 @@ func LookUpStorageSpacesForPathWithChildren(ctx context.Context, client gateway.
 	}
 
 	lSSRes, err := client.ListStorageSpaces(ctx, lSSReq)
-	if err != nil || lSSRes.Status.Code != rpc.Code_CODE_OK {
+	if err != nil {
+		return nil, nil, err
+	}
+	if lSSRes == nil {
+		return nil, &rpc.Status{Code: rpc.Code_CODE_INTERNAL, Message: "no error but result was nil"}, nil
```

Comment:
This should NEVER happen. If we need to check it, please log it with highest priority so we are aware of it and can fix it.

Reply:
when `err != nil` then `lSSRes.Status` will nill because `lSSRes` might be nil ;-)

Reply:
it will all go away when we have a service registry ... for it is good enough to return the errors properly.

Reply:
> when `err != nil` then `lSSRes.Status` will nill because `lSSRes` might be nil ;-)

But if `err != nil` then `lSSRes.Status.Code != rpc.Code_CODE_OK` would never be executed.
If `err == nil` and `lSSRes == nil` too then something in `ListStorageSpaces` is wrong.

Reply:
That's exactly what I meant. In my opinion if a handler returns an `err == nil` then the result MUST never be nil so I can check the status code.

Reply:
I agree that `lssRes` should never be nil when `err == nil` but the `Status` field could still be nil which would make the `lssRes.Status.Code` access fail, , maybe that's what @butonic was seeing?
We can't expect `lssRes.Status` to never be nil in the no-error case I think, so how about changing the check to `if lssRes.Status == nil`?

Reply:
Actually not needed. `lssRes.Status.GetCode` will return `rpc.Code_CODE_INVALID` even if `lssRes.Status` is `nil`.

Reply:
That's even better :) :+1:

Reply:
@kobergj could you suggest a code change?

Reply:
@C0rby @aduffeck I removed the line. Can some of you approve this? My approve did go away because I was pushing...